app/models/profile_field.rb

```diff
@@ -0,0 +1,12 @@
+class ProfileField < ApplicationRecord
+  # Key names follow the Rails form helpers
+  enum input_type: {
+    text_field: 0,
+    text_area: 1,
+    check_box: 2,
+    color_field: 3
+  }
+
+  validates :label, presence: true, uniqueness: { case_sensitive: false }
```

Comment:
This column uses Postgres' case-insensitive text type (`citext`) so the case insensitive uniqueness is also ensured when we don't go through the model:

```
PracticalDeveloper_development=# select label from profile_fields;
 label
-------
 Email
(1 row)

PracticalDeveloper_development=# insert into profile_fields (label, created_at, updated_at) values ('email', now(), now());
ERROR:  duplicate key value violates unique constraint "index_profile_fields_on_label"
DETAIL:  Key (label)=(email) already exists.
```

This seems better than a functional index and is also supported by the schema DSL. Rails natively supports `citext` since 4.2 btw, so this is not some new experimental feature.

Reply:
TIL!